include/linux/kfence.h

```diff
@@ -23,31 +22,118 @@ extern char __kfence_pool[KFENCE_POOL_SIZE];
+ *
+ * Return: true or false depending on whether the address is within the KFENCE object range.
+ *
+ * KFENCE objects live in a separate page range and are not to be intermixed with regular heap
+ * objects (e.g. KFENCE objects must never be added to SL[AU]B freelists). Failing to do so may
+ * and will result in heap corruptions, therefore is_kfence_address() must be used to check whether
+ * an object requires specific handling.
+ */
 static __always_inline bool is_kfence_address(const void *addr)
 {
 	return unlikely((char *)addr >= __kfence_pool &&
 			(char *)addr < __kfence_pool + KFENCE_POOL_SIZE);
 }
 
+/**
+ * kfence_init() - perform KFENCE initialization at boot time.
+ */
 void kfence_init(void);
 
+/**
+ * kfence_shutdown_cache() - handle shutdown_cache() for KFENCE objects.
+ * @s: cache being shut down.
+ *
+ * Return: true on success, false if any leftover objects persist.
+ *
+ * Before shutting down a cache, one must ensure there are no remaining objects allocated from it.
+ * KFENCE objects are not referenced from the cache, so kfence_shutdown_cache() takes care of them.
+ */
 bool __must_check kfence_shutdown_cache(struct kmem_cache *s);
 
+
+/**
+ * __kfence_alloc() - allocate a KFENCE object.
+ * @s:     struct kmem_cache with object requirements.
+ * @size:  exact size of the object to allocate (can be less than @s->size e.g. for kmalloc caches).
+ * @flags: GFP flags.
+ *
+ * Return: pointer to a KFENCE object, if available, NULL otherwise.
+ *
+ * __kfence_alloc() performs a deterministic allocation from the KFENCE pool using the requested
+ * parameters.
+ */
 void *__kfence_alloc(struct kmem_cache *s, size_t size, gfp_t flags);
 
-// TODO(elver): Add API doc.
+/**
+ * kfence_alloc() - allocate a KFENCE object with a low probability.
+ * @s:     struct kmem_cache with object requirements.
+ * @size:  exact size of the object to allocate (can be less than @s->size e.g. for kmalloc caches).
+ * @flags: GFP flags.
+ *
+ * Return:
+ * * NULL     - must proceed with allocating from SL[AU]B as usual.
+ * * non-NULL - pointer to a KFENCE object.
+ *
+ * kfence_alloc() should be inserted into the SL[AU]B allocation fast path, allowing it to
+ * transparently return KFENCE-allocated objects with a low probability using a static branch
+ * (the probability is controlled by the kfence.sample_interval boot parameter).
+ */
 static __always_inline void *kfence_alloc(struct kmem_cache *s, size_t size, gfp_t flags)
 {
 	return static_branch_unlikely(&kfence_allocation_key) ? __kfence_alloc(s, size, flags) :
 								      NULL;
 }
 
+/**
+ * kfence_ksize() - get the actual amount of memory allocated for a KFENCE object.
+ * @addr: pointer to a heap object.
+ *
+ * Return:
+ * * 0     - not a KFENCE object, must call __ksize() instead.
+ * * non-0 - this many bytes can be accessed without causing a memory error.
+ *
+ * ksize() returns the number of bytes past the end of a SL[AU]B object that can be safely used
```

Comment:
I would lead with explicitly saying what kfence_ksize() does. And if you want, conclude with the fact that it mirrors ksize()'s behaviour (but leading with an explanation of ksize() makes it look like this is ksize()'s documentation).

Reply:
Fixed.